File: src/main/webapp/js/service/studentService.js
//服务层
app.service('studentService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../student/findAll');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../student/findPage?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../student/findOne?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../student/add',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../student/update',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../student/delete?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../student/search?page='+page+"&rows="+rows, searchEntity);
	}
	//下拉列表数据
	this.selectOptionList=function(){
		return $http.get('../student/selectOptionList');
	}
});

File: src/main/java/com/student/service/StudentService.java
package com.student.service;
import java.util.List;
import java.util.Map;

import com.student.pojo.Student;

import entity.PageResult;
/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface StudentService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<Student> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(Student student);
	
	
	/**
	 * 修改
	 */
	public void update(Student student);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public Student findOne(int id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(int [] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(Student student, int pageNum,int pageSize);

	/**
	 * 分页测试
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPageTest();
	/**
	 * 下拉框查询
	 * @return
	 */
	public List<Map> selectOptionList();
	
}
